### ISSUE TITLE
Using same DatasetGraph for updates and query throws an INVALIDPARAM error

### ISSUE THREAD

**Issue by grechaw** [marklogic/marklogic-jena#62]
I've set up a multi-threaded scenario which does both updates and queries.  This situation runs straight into a change in the Java Client API regarding threading:

from SPARQLQueryManager Javadoc:

```
 * <p>SPARQLQueryManager is thread-safe other than setPageLength. In other
 * words the only state maintained by an instance is the page length.  Common
 * usage is to call setPageLength only once then use the instance across many
 * threads.  If you intend to call setPageLength from multiple threads, create
 * a new SPARQLQueryManager for each thread.</p>
```

What happens is that, after one has used a querymanager for a select, then use the same one for an update, the pageLength parameter is sent along with an update and fails.

This parameter was ignored until a change in /v1/graphs/sparql a couple of weeks ago.
Before that, pageLength was ignored by SPARQL update queries.

**Comment by grechaw** [comment]
This bug only surfaces against a nightly build of MarkLogic Server.  It is a regression that happened sometime in August 2017.

A fix could be built into marklogic-jena, but I think there's a good case to make the fix in the server, so as to avoid the incompatibility we've built into the client.

**Comment by grechaw** [comment]
@kkanthet, this issue is lingering from a previous release.  Can you confirm that you saw and ship?

**Comment by kkanthet** [comment]
Yes this issue is shipped in 3.0.4 , closing it.
